landlab/components/erosion_deposition/erosion_deposition.py

```diff
@@ -172,7 +182,10 @@ def __init__(
         K : float, field name, or array
             Erodibility for substrate (units vary).
         phi : float
-            Sediment porosity [-].
+            Sediment porosity [-]. NOTE: This parameter is retained for backward
+            compatibility, but has no impact on the solution because all
+            calculations are performed in terms of bulk equivalent volume
+            or depth.
```

Comment:
@gregtucker I think all your changes are great. the only thing I would change is that I would argue that we should remove phi here and as something in the keyword arguments because it is not an input. We can make a note in the docstring that says that it previously was an option, but now it is not. I think leaving it will get people into the same trouble that **kwds did. 

I agree that we wan't to do as little as possible to make LL2 break pre LL2 code, but I think this is a case in which it should. If a user was using a pre v2.0 version of ErosionDeposition and they were passing phi, it did something. Now it doesn't. So no matter what we aren't maintaining backward compatibility.

However, I think this could be a really good opportunity to use **kwds. Specifically, we add **kwds, and if phi is passed by a user, we raise an error with an informative message indicating how and why this component is changed. This means that if a user had code that used phi as a keyword argument, we are making it easy for them to know what has changed (they don't have to dig through the source code, etc). 

I'm happy to implement this, but will wait to hear what you think.

Reply:
@kbarnhart you make a fair point about breaking backward compatibility: we've already done it here, so might as well go all the way. More honest to have the component complain than to have it produce slightly different numbers. 

**kwds seems like a reasonable solution *if* we also guard against passing anything else, such as valid parameters that are just mis-spelled. In other words, **kwds needs to be empty or just contain phi. Do you want me to implement? Having now finished the above stuff sooner than I thought would be possible, I have some time.

Reply:
I'm on it. But I think it would be best if you looked at my error text to make sure it is as comprehensible as possible.